tests/integration/standard/test_connection.py

```diff
@@ -41,6 +41,84 @@ def setup_module():
     use_singledc()
 
 
+class TestHostListener(HostStateListener):
+    host_down = None
+
+    def on_down(self, host):
+        host_down = host
+
+
+class HeartbeatTest(unittest.TestCase):
+    """
+    Test to validate failing a heartbeat check doesn't mark a host as down
+
+    @since 3.3
+    @jira_ticket PYTHON-286
+    @expected_result host should not be marked down when heartbeat fails
+
+    @test_category connection heartbeat
+    """
+
+    def setUp(self):
+        self.cluster = Cluster(protocol_version=PROTOCOL_VERSION, idle_heartbeat_interval=1)
+        self.session = self.cluster.connect()
+
+    def tearDown(self):
+        self.cluster.shutdown()
+
+    def test_heart_beat_timeout(self):
+        # Setup a host listener to ensure the nodes don't go down
+        test_listener = TestHostListener()
+        host = "127.0.0.1"
+        node = get_node(1)
+        initial_connections = self.fetch_connections(host, self.cluster)
+        self.assertNotEqual(len(initial_connections), 0)
+        self.cluster.register_listener(test_listener)
+        # Pause the node
+        node.pause()
+        # Wait for connections associated with this host go away
+        self.wait_for_no_connections(host, self.cluster)
+        # Resume paused node
+        node.resume()
+        # Run a query to ensure connections are re-established
+        self.session.execute("SELECT * FROM system.local")
```

Comment:
Might be more rigorous to run until the queried host is the one that was paused.